tests/testthat/test-grid_helpers.R

```diff
@@ -185,25 +189,30 @@ test_that("compute_grid_info - recipe and model (with and without submodels)", {
   # use grid_regular to (partially) trigger submodel trick
   set.seed(1)
   param_set <- extract_parameter_set_dials(wflow)
-  grid <- bind_rows(grid_regular(param_set), grid_space_filling(param_set))
+  grid <-
+    bind_rows(grid_regular(param_set), grid_space_filling(param_set)) %>%
+    arrange(deg_free, loss_reduction, trees)
   res <- compute_grid_info(wflow, grid)
 
   expect_equal(length(unique(res$.iter_preprocessor)), 5)
   expect_equal(
     unique(res$.msg_preprocessor),
     paste0("preprocessor ", 1:5, "/5")
   )
-  expect_equal(res$trees, c(rep(max(grid$trees), 10), 1))
+  expect_equal(sort(res$trees), sort(c(rep(max(grid$trees), 10), 1)))
   expect_equal(unique(res$.iter_model), 1:3)
   expect_equal(
-    res$.iter_config[1:3],
+    res$.iter_config[res$.iter_preprocessor == 1],
     list(
```

Comment:
Your new enumeration is correct (and fixes the case where I had broken previously), but it does look like we're missing a `format_with_padding` still. When I run these tests with CRAN `compute_grid_info()`, I see:

```
── Failure (test-grid_helpers.R:204:3): compute_grid_info - recipe and model (with and without submodels) ──
res$.iter_config[res$.iter_preprocessor == 1] (`actual`) not equal to list(...) (`expected`).

    actual[[1]]             | expected[[1]]             
[1] "Preprocessor1_Model01" - "Preprocessor1_Model1" [1]
[2] "Preprocessor1_Model02" - "Preprocessor1_Model2" [2]
[3] "Preprocessor1_Model03" - "Preprocessor1_Model3" [3]
[4] "Preprocessor1_Model04" - "Preprocessor1_Model4" [4]

`actual[[2]]`:   "Preprocessor1_Model05" "Preprocessor1_Model06" "Preprocessor1_Model07"
`expected[[2]]`: "Preprocessor1_Model5"  "Preprocessor1_Model6"  "Preprocessor1_Model7" 

`actual[[3]]`:   "Preprocessor1_Model08" "Preprocessor1_Model09" "Preprocessor1_Model10"
`expected[[3]]`: "Preprocessor1_Model8"  "Preprocessor1_Model9"  "Preprocessor1_Model10"
```